docs/DeveloperGuide.md

```diff
@@ -4,22 +4,194 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Design
+
+#### Architecture
+
+{Description of overall architecture}
+
+#### Ui Component
+
+The structure of the Ui Component is illustrated in the following diagram:
+
+![UiComponentDiagram](images/UiComponent.png)
```

Comment:
should the relationship between abstract command class and its child command classes be inheritance instead of dependency?